### ISSUE TITLE
Chef never successfully completed!

### ISSUE THREAD

**Issue by dalareo** [p2pu/lernanta-dev-env#11]
==> default: Error executing action `run` on resource 'bash[Make Database]'
==> default: ================================================================================
==> default: Mixlib::ShellOut::ShellCommandFailed
==> default: ------------------------------------
==> default: Expected process to exit with [0], but received '1'
==> default: ---- Begin output of "bash"  "/tmp/chef-script20160112-4311-rzc65l-0" ----
==> default: STDOUT: 
==> default: STDERR: Error: cannot import name setup
==> default: ---- End output of "bash"  "/tmp/chef-script20160112-4311-rzc65l-0" ----
==> default: Ran "bash"  "/tmp/chef-script20160112-4311-rzc65l-0" returned 1
==> default: 
==> default: Resource Declaration:
==> default: ---------------------
==> default: # In /tmp/vagrant-chef-5/chef-solo-1/cookbooks/main/recipes/lernanta.rb
==> default: 
==> default: 124: 
==> default: 125: bash "Make Database" do
==> default: 126:   # user "p2pu"
==> default: 127:   # group "p2pu"
==> default: 128:   code <<-EOH
==> default: 129:     sudo su -l -c 'workon lernanta && python /opt/lernanta/lernanta/manage.py syncdb --noinput' p2pu
==> default: 130:   EOH
==> default: 131: end
==> default: 132: 
==> default: 
==> default: Compiled Resource:
==> default: ------------------
==> default: # Declared in /tmp/vagrant-chef-5/chef-solo-1/cookbooks/main/recipes/lernanta.rb:125:in `from_file'
==> default: 
==> default: bash("Make Database") do
==> default:   retry_delay 2
==> default:   returns 0
==> default:   command "\"bash\"  \"/tmp/chef-script20160112-4311-rzc65l-0\""
==> default:   retries 0
==> default:   recipe_name "lernanta"
==> default:   code "    sudo su -l -c 'workon lernanta && python /opt/lernanta/lernanta/manage.py syncdb --noinput' p2pu\n"
==> default:   action "run"
==> default:   cookbook_name :main
==> default:   interpreter "bash"
==> default:   backup 5
==> default: end
==> default: 
==> default: [2016-01-12T13:00:51+00:00] ERROR: Running exception handlers
==> default: [2016-01-12T13:00:51+00:00] ERROR: Exception handlers complete
==> default: [2016-01-12T13:00:51+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> default: [2016-01-12T13:00:51+00:00] FATAL: Mixlib::ShellOut::ShellCommandFailed: bash[Make Database](main::lernanta line 125) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
==> default: ---- Begin output of "bash"  "/tmp/chef-script20160112-4311-rzc65l-0" ----
==> default: STDOUT: 
==> default: STDERR: Error: cannot import name setup
==> default: ---- End output of "bash"  "/tmp/chef-script20160112-4311-rzc65l-0" ----
==> default: Ran "bash"  "/tmp/chef-script20160112-4311-rzc65l-0" returned 1

**Comment by dirkcuys** [comment]
Hi @dalareo - it's been ages since I've used chef to setup Lernanta. Atm lernanta is mostly in maintenance, but you may have better luck using this Ansible playbook: https://github.com/p2pu/marvin/tree/master/lernanta.

Are you interested in running your own version of lernanta?

**Comment by dalareo** [comment]
Thanks @dirkcuys, I just was exploring Ansible playbooks right now. I am exploring Lernanta as a light alternative to edX. Have you dockerize it?

**Comment by dirkcuys** [comment]
No, not yet. If you are looking for something for a single course, you should look at https://github.com/p2pu/course-in-a-box - This is what we are using for any new courses. Here are two examples of courses using Course in a Box:
1. http://writing4change.p2pu.org/
2. http://python.p2pu.org/

**Comment by dalareo** [comment]
But is Lernanta still used in P2PU? 
I am designing a digital platform for an education institution on top of a Container as a service archictecture using Rancher/Docker. Your jekyll-approach looks very interesting but I also need a more integrated one like edX or Lernanta.
Making courses almost portable and hosting them in static servers is very interesting. I am also exploring static-sites for other things inside the same project, like team-blogs and digital-products launching.
The education institution I am working with deals with team-enterpreneuring, related to Team Academy methodology. Modular and scalable solutions are the "core" of my proposal.

**Comment by dirkcuys** [comment]
We've shifted focus to helping people take online courses rather than helping people create online courses. To that end we are working with libraries to host in-person study groups working through online courses hosted on any online platform.

Lernanta still runs at courses.p2pu.org & we have a soft spot for people interested in creating peer driven online courses, but lernanta is not in active development.

If I wanted to position lernanta as a scaled down version of EdX, I would cut a lot of the legacy parts and keep it very focussed. But that would require some development.
